Update vignette

ldecicco-USGS opened the pull request.

ldecicco-USGS committed:
Update readme with new flow

ldecicco-USGS committed:
Merge branch 'add_docs' into 'develop'

Fix pkgdown

See merge request water/dataRetrieval!445

ldecicco-USGS committed:
Merge branch 'add_docs' into 'develop'

fix NLDI test

See merge request water/dataRetrieval!446

ldecicco-USGS committed:
make sure this works on gitlab

ldecicco-USGS committed:
Remove bbox from daily

ldecicco-USGS committed:
Merge branch 'develop' of github.com:DOI-USGS/dataRetrieval into update_vignette

ldecicco-USGS committed:
Add API key info to readme

ldecicco-USGS committed:
add discrete stuff

ldecicco-USGS committed:
Merge branch 'develop' of https://code.usgs.gov/water/dataRetrieval into update_vignette

ldecicco-USGS committed:
put bbox back in daily

ldecicco-USGS committed:
Vignette updates

ldecicco-USGS commented:
@mikemahoney218-usgs @jzemmels @ehinman @cnell-usgs 
Here's the in-development rendered doc:
https://water.code-pages.usgs.gov/dataRetrieval/articles/read_USGS_functions.html

Any feedback would be very much appreciated. I'm hoping this document stays in active development, so this is just a first draft to introduce folks to the functions.

ldecicco-USGS committed:
Merge branch 'develop' of github.com:DOI-USGS/dataRetrieval into update_vignette

ldecicco-USGS committed:
Add help section

cnell-usgs commented on vignettes/read_USGS_functions.Rmd:
````diff
@@ -24,151 +24,227 @@ options(continue = " ",
 
-* Flexible data/last_modified
+```{r}
+library(leaflet)
+leaflet_crs <- "+proj=longlat +datum=WGS84" #default leaflet crs
````
```suggestion
library(leaflet)

leaflet_crs <- "+proj=longlat +datum=WGS84" #default leaflet crs

```

cnell-usgs commented on vignettes/read_USGS_functions.Rmd:
````diff
@@ -24,151 +24,227 @@ options(continue = " ",
 ```{r}
-sites_modern <- read_USGS_monitoring_location(monitoring_location_id = "USGS-01491000")
-names(sites_modern)
+library(ggplot2)
+ggplot(data = daily_modern) +
````
```suggestion
library(ggplot2)

ggplot(data = daily_modern) +
```

cnell-usgs commented on vignettes/read_USGS_functions.Rmd:
````diff
@@ -24,151 +24,227 @@ options(continue = " ",
+Let's see what that looks like:
+
+```{r}
+leaflet_crs <- "+proj=longlat +datum=WGS84" #default leaflet crs
+pal <- colorNumeric("viridis", sites_mn_wi$drainage_area)
+leaflet(data = sites_mn_wi |> 
````
```suggestion
leaflet_crs <- "+proj=longlat +datum=WGS84" #default leaflet crs

pal <- colorNumeric("viridis", sites_mn_wi$drainage_area)

leaflet(data = sites_mn_wi |> 
```

jzemmels commented on vignettes/read_USGS_functions.Rmd:
````diff
@@ -24,151 +24,227 @@ options(continue = " ",
+The returned data includes a column "geometry" which is a collection of simple feature (sf) points. This allows for seamless integration with the `sf` package. Here are 2 quick examples of using the `sf` object in ggplot2 and leaflet:
+
+```{r}
+library(ggplot2)
````
SC: `sf` is [suggested](https://github.com/tidyverse/ggplot2/blob/main/DESCRIPTION#L69), so installation may be required for new users.
```suggestion
# install.packages("sf")
library(ggplot2)
```

ldecicco-USGS replied:
We added `sf` as an Import (before it was just a suggestion), so presumably any user who uses the new version should have it. So, I'm not sure this is a great place for that, but I _should_ update the Dependency section below (when I first started writing this up we were still debating keeping it as suggest vs import.)

The issue I've seen with including commented out install instructions like this is that folks tend to uncomment it and have consistently have fresh installs of that one package (in this case "sf"), and then CRAZY outdated packages everywhere else.

ldecicco-USGS committed:
Update vignettes/read_USGS_functions.Rmd

Co-authored-by: Cee Nell <cnell@usgs.gov>

ldecicco-USGS committed:
Update vignettes/read_USGS_functions.Rmd

Co-authored-by: Cee Nell <cnell@usgs.gov>

ldecicco-USGS committed:
Update vignettes/read_USGS_functions.Rmd

Co-authored-by: Cee Nell <cnell@usgs.gov>

jzemmels reviewed (commented):
Looks great. A few suggested changes.

ldecicco-USGS committed:
Update vignettes/read_USGS_functions.Rmd

Co-authored-by: Joe Zemmels (he/him) <jzemmels@gmail.com>

ldecicco-USGS committed:
Update vignettes/read_USGS_functions.Rmd

Co-authored-by: Joe Zemmels (he/him) <jzemmels@gmail.com>

ldecicco-USGS committed:
Update vignettes/read_USGS_functions.Rmd

Co-authored-by: Joe Zemmels (he/him) <jzemmels@gmail.com>

ldecicco-USGS committed:
Update vignettes/read_USGS_functions.Rmd

Co-authored-by: Joe Zemmels (he/him) <jzemmels@gmail.com>

mikemahoney218-usgs reviewed (commented):
Super exciting seeing this!

ehinman reviewed (commented):
This looks and reads very well, @ldecicco-USGS! I especially loved the CQL section. Seems very cool. Made a few minor wording suggestions, plus a question about the term "NWIS".

ehinman commented on vignettes/read_USGS_functions.Rmd:
```diff
@@ -24,151 +24,227 @@ options(continue = " ",
 
 
-As we bid adieu to the NWIS web services, we welcome a new web service offering, the USGS Water Data OGC APIs.
+As we bid adieu to the NWIS web services, we welcome a host of new web service offering: the [USGS Water Data OGC APIs](https://api.waterdata.usgs.gov/ogcapi/v0/). This is a modern access point for USGS water data. The USGS is planning to modernize **all** web services in the near future. For each of these updates, `dataRetrieval` will create a new function to access the new services. 
```
A simple fix might be to replace NWIS with "WaterServices".

mikemahoney218-usgs replied:
NWIS is the broadest possible umbrella containing almost the entire water observing system. Our group owns "NWIS Delivery", which covers WDFN, WQP, WaterAlert, STN, RT-FIM, NGWMN, NWD, comptools, the new APIs, and so on. But NWIS itself also includes AQTS, AQS, MLR, NIMS, TID, DECAP, DIRS, RLMS... there's a few dozen other acronyms included, everything that's required to keep the data flowing. We'll be continuing to call this system NWIS going forward (the replacement for the current NWISMod will be called NWIS). The purpose of api.waterdata.usgs.gov is to be "a catalog of all modernized NWIS APIs" going forward -- that's why it includes now RTFI-API, for instance.

That said, we're moving away from tagging each of the subcomponents as "NWIS" -- rather than "NWISWeb" and "NWISMapper", it's now "WDFN" and "Explore USGS Water Data" and "How to access USGS data in QGIS" and so on. 

Maybe that answers the question?

ldecicco-USGS committed:
Update README.Rmd

Co-authored-by: Elise Hinman <121896266+ehinman@users.noreply.github.com>

ldecicco-USGS committed:
Update vignettes/read_USGS_functions.Rmd

Co-authored-by: Elise Hinman <121896266+ehinman@users.noreply.github.com>

ldecicco-USGS committed:
Update vignettes/read_USGS_functions.Rmd

Co-authored-by: Cee Nell <cnell@usgs.gov>

ldecicco-USGS commented on vignettes/read_USGS_functions.Rmd:
```diff
@@ -24,151 +24,227 @@ options(continue = " ",
+Each new "API endpoint" will deliver a new type of USGS data. Currently the available endpoints are "monitoring-locations", "time-series-metadata", and "daily". All of these new endpoints offer some new features that the NWIS services did not have:
 
-`r dataRetrieval:::get_description("daily")`
+* Any column that is returned can also be queried on! The main `dataRetrieval` functions will take vector inputs. Each query is appended as a boolean AND. For instance, if you ask for a vector of monitoring location ids and a vector of parameter codes, you will get data back for just those monitoring locations AND the specified parameter codes. Towards the bottom of this document, we'll discuss w ...
```
I'll work on a re-write to hopefully clarify. It's not necessarily unique to this service - but it is unique to the way dataRetrieval does it (because the API is expecting the CQL which is way more flexible than a bunch of ANDs)

mikemahoney218-usgs commented on vignettes/read_USGS_functions.Rmd:
```diff
@@ -24,151 +24,232 @@ options(continue = " ",
+The `read_USGS_data` function replaces the `readNWISdata` function. This is a lower-level, generalized function for querying any of the API endpoints.
 
-If you find yourself running into limits, you can request an API token here: <https://api.waterdata.usgs.gov/signup/>
+The new APIs can handle complex requests. For those queries, users will need to construct their own request using Common Query Language. There's an excellent article <https://api.waterdata.usgs.gov/docs/ogcapi/complex-queries/>.
```
So here's a weird one... we use CQL2. Which OGC always refers to as "Common Query Language" and never calls "version 2". Not sure if it's worth specifying "(CQL2)" after the full name is spelled out?

https://www.ogc.org/standards/cql2/

ldecicco-USGS committed:
some updates from reviewers

ldecicco-USGS committed:
pull from PR

Merge branch 'update_vignette' of github.com:ldecicco-USGS/dataRetrieval into update_vignette

# Conflicts:
#	vignettes/read_USGS_functions.Rmd

ldecicco-USGS committed:
vignette updates
